In [46]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import datetime

In [64]:
#  Creando dataFrames de CSV a analizar
actividad_diaria = pd.read_csv('dailyActivity_merged.csv')
sleep_day = pd.read_csv('sleepDay_merged.csv')

In [66]:
# Transformando a horas el tiempo de sueño
sleep_day['hours_sleep'] = round((sleep_day['TotalTimeInBed'] / 60),2) 

In [67]:
# Renombrando la columna de la fecha para posteriormente hacer merged
sleep_day.rename(columns={'SleepDay':'ActivityDate'}, inplace=True)

In [68]:
# Hay 3 filas duplicadas, las eliminamos 
sleep_day = sleep_day.drop_duplicates()

In [69]:
# Columnas mas relevantes
# actividad_diaria3 = actividad_diaria[['Id','TotalSteps','TotalDistance','VeryActiveDistance','LightActiveDistance','SedentaryActiveDistance','Calories']]

In [70]:
# Transformando en DateTime la fecha
actividad_diaria['ActivityDate'] = pd.to_datetime(actividad_diaria['ActivityDate'])

In [71]:
# Transformando en DateTime la fecha
sleep_day['ActivityDate'] =  pd.to_datetime(sleep_day['ActivityDate'])

C:\Users\Abdon.RAMIREZ-BRICEN\AppData\Local\Temp\ipykernel_32560\187503775.py:2: UserWarning: Could not infer format, so each element will be parsed individually, falling back to `dateutil`. To ensure parsing is consistent and as-expected, please specify a format.
  sleep_day['ActivityDate'] =  pd.to_datetime(sleep_day['ActivityDate'])


In [72]:
# Agregando al DataFrame principal la columna de Horas de sueño
activity = pd.merge(actividad_diaria,sleep_day[['Id','ActivityDate','hours_sleep']],on=['Id','ActivityDate'],how='left')

In [73]:
# Elimianndo filas duplicadas
activity = activity.drop_duplicates()

In [74]:
# Guardamos en csv el DataFrame actualizado
activity.to_csv('activity_users.csv')
activity.to_json('json_activity.json')

In [75]:
pd.set_option('display.max_rows', None)  # Mostrar todas las filas
pd.set_option('display.max_columns', None)  # Mostrar todas las columnas

In [76]:
a = activity.groupby(['Id', 'ActivityDate'])['hours_sleep'].value_counts().reset_index()

## Comenzamos con el analisis.

In [77]:
activity.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 940 entries, 0 to 939
Data columns (total 16 columns):
 #   Column                    Non-Null Count  Dtype         
---  ------                    --------------  -----         
 0   Id                        940 non-null    int64         
 1   ActivityDate              940 non-null    datetime64[ns]
 2   TotalSteps                940 non-null    int64         
 3   TotalDistance             940 non-null    float64       
 4   TrackerDistance           940 non-null    float64       
 5   LoggedActivitiesDistance  940 non-null    float64       
 6   VeryActiveDistance        940 non-null    float64       
 7   ModeratelyActiveDistance  940 non-null    float64       
 8   LightActiveDistance       940 non-null    float64       
 9   SedentaryActiveDistance   940 non-null    float64       
 10  VeryActiveMinutes         940 non-null    int64         
 11  FairlyActiveMinutes       940 non-null    int64         
 12  LightlyActiveMinutes  

In [78]:
activity.shape

(940, 16)

In [79]:
activity.columns

Index(['Id', 'ActivityDate', 'TotalSteps', 'TotalDistance', 'TrackerDistance',
       'LoggedActivitiesDistance', 'VeryActiveDistance',
       'ModeratelyActiveDistance', 'LightActiveDistance',
       'SedentaryActiveDistance', 'VeryActiveMinutes', 'FairlyActiveMinutes',
       'LightlyActiveMinutes', 'SedentaryMinutes', 'Calories', 'hours_sleep'],
      dtype='object')

In [80]:
activity.isnull().sum()

Id                            0
ActivityDate                  0
TotalSteps                    0
TotalDistance                 0
TrackerDistance               0
LoggedActivitiesDistance      0
VeryActiveDistance            0
ModeratelyActiveDistance      0
LightActiveDistance           0
SedentaryActiveDistance       0
VeryActiveMinutes             0
FairlyActiveMinutes           0
LightlyActiveMinutes          0
SedentaryMinutes              0
Calories                      0
hours_sleep                 530
dtype: int64

Observamos que tenemos 530 datos nulos en las horas de sueño. Trataremos de realizar un analisis mas adelante con los datos que tenemos.